// components/searchPrice/index.js
import React, {Component} from 'react'
import {connect} from 'react-redux'

import {searchPrice} from 'actions'

class SearchPrice extends Component {
    constructor(props) {
        super(props)

        this.state = {
            min: '1000',
            max: '95000'
        }

        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleChangeMin = (event) => {
        this.setState({
            min: event.target.value
        })
    }

    handleChangeMax = (event) => {
        this.setState({
            max: event.target.value
        })
    }

    handleSubmit(event) {
        event.preventDefault()
        this.props.searchPrice(this.state)
    }

    render() {
        return (
            <div className='well blosd'>
                <h3 className='lead'>Цена</h3>
                <div>
                    <form onSubmit={this.handleSubmit} className='price-filter'>
                        <label>От</label>
                        <input
                            onChange={this.handleChangeMin}
                            type='text'
                            className='form-control'
                            value={this.state.min}
                        />
                        <label>До</label>
                        <input
                            onChange={this.handleChangeMax}
                            type='text'
                            className='form-control'
                            value={this.state.max}
                        />
                        <span className=''>
            <button className='btn btn-red btn-price-filter btn-block' type='submit'>
                Применить
            </button>
          </span>
                    </form>

                </div>
            </div>
        )
    }
}

const mapDispatchToProps = {
    searchPrice
}

export default connect(null, mapDispatchToProps)(SearchPrice)
