// components/subscribe/index.js
import React, {Component} from 'react'


class Subscribe extends Component {
    constructor(props) {
        super(props)

        this.state = {
            value: ''
        }

        this.handleChange = this.handleChange.bind(this)
        this.handleSubmit = this.handleSubmit.bind(this)
    }

    handleChange(event) {
        console.log(event.target.value)
        this.setState({
            value: event.target.value
        })
    }

    handleSubmit(event) {
        event.preventDefault()

        const request = new XMLHttpRequest();
        request.open('POST', 'https://telephone.su/api/subscribe/add.php');

        let obj = {}
        obj['email'] = this.state.value;

        let json = JSON.stringify(obj);
        request.send(json);
        console.log('json', json)
        let statusMessage = document.querySelector('.tyForSub')
        request.addEventListener('readystatechange', () => {
            statusMessage.innerHTML = 'Спасибо за подписку!';
        });

    }

    render (){
        return (
            <div className="section newsletter">

                <div className="container">

                    <div className="row">
                        <div className="col-md-12">
                            <div className="newsletter">
                                <p className='newsletter-text'>Подпишитесь на <strong>рассылку</strong> что бы всегда узнавать об акциях и о новом товаре!</p>
                                <form onSubmit={this.handleSubmit} method='POST' className='sub-form'>
                                    <input className="input" type="email" placeholder="введите ваш email" onChange={this.handleChange} />
                                        <button className="newsletter-btn"><i className="fa fa-envelope"></i> подписаться
                                        </button>
                                </form>
                                <div className='tyForSub'></div>
                                <ul className="newsletter-follow">
                                    <p>Мы в <strong>социальных сетях</strong>!</p>
                                    <li>
                                        <a href="/#"><i className="fa fa-facebook"></i></a>
                                    </li>
                                    <li>
                                        <a href="/#"><i className="fa fa-telegram"></i></a>
                                    </li>
                                    <li>
                                        <a href="/#"><i className="fa fa-instagram"></i></a>
                                    </li>
                                    <li>
                                        <a href="/#"><i className="fa fa-vk"></i></a>
                                    </li>
                                </ul>
                            </div>
                        </div>
                    </div>
                </div>

            </div>

        )
    }
}

export default Subscribe
